pose-detection/src/posenet/detector.ts

```diff
@@ -161,6 +162,91 @@ class PosenetDetector implements PoseDetector {
     return scaledPoses;
   }
 
+  async estimateSinglePoseGPU(
+      image: PoseDetectorInput,
+      estimationConfig:
+          PoseNetEstimationConfig = SINGLE_PERSON_ESTIMATION_CONFIG):
+      Promise<[tf.Tensor[], number[]]> {
+    const config = validateEstimationConfig(estimationConfig);
+
+    if (image == null) {
+      return [[], []];
+    }
+
+    this.maxPoses = config.maxPoses;
+
+    const {imageTensor, padding} = convertImageToTensor(image, {
+      outputTensorSize: this.inputResolution,
+      keepAspectRatio: true,
+      borderMode: 'replicate'
+    });
+
+    const imageValueShifted = this.architecture === 'ResNet50' ?
+        tf.add(imageTensor, RESNET_MEAN) :
+        shiftImageValue(imageTensor, [-1, 1]);
+
+    const results =
+        this.posenetModel.predict(imageValueShifted) as tf.Tensor4D[];
+
+    let offsets, heatmap, displacementFwd, displacementBwd;
+    if (this.architecture === 'ResNet50') {
+      offsets = tf.squeeze(results[2], [0]);
+      heatmap = tf.squeeze(results[3], [0]);
+      displacementFwd = tf.squeeze(results[0], [0]);
+      displacementBwd = tf.squeeze(results[1], [0]);
+    } else {
+      offsets = tf.squeeze(results[0], [0]);
+      heatmap = tf.squeeze(results[1], [0]);
+      displacementFwd = tf.squeeze(results[2], [0]);
+      displacementBwd = tf.squeeze(results[3], [0]);
+    }
+    const heatmapScores = tf.sigmoid(heatmap) as tf.Tensor3D;
+
+    let poses;
+
+    if (this.maxPoses === 1) {
+      const pose = await decodeSinglePoseGPU(
+          heatmapScores, offsets as tf.Tensor3D, this.outputStride);
+      poses = [pose];
+    } else {
+      throw new Error('GPU renderer only supports single pose!');
+    }
+    // TODO: handle flipPosesHorizontal in GPU.
+    if (config.flipHorizontal === true) {
+      throw new Error('flipHorizontal is not supported!');
+    }
+
+    const canvasInfo =
+        this.getCanvasInfo(getImageSize(image), this.inputResolution, padding);
+    imageTensor.dispose();
+    imageValueShifted.dispose();
+    tf.dispose(results);
+    offsets.dispose();
+    heatmap.dispose();
+    displacementFwd.dispose();
+    displacementBwd.dispose();
+    heatmapScores.dispose();
+
+    return [poses, canvasInfo];
```

Comment:
Add a todo here to indicate that `score` in not supported yet.

Reply:
Add todo in decodeSinglePoseGPU